### ISSUE TITLE
Apple silicon M3 Pro inference wrong,  output video are all with black masked frame

### ISSUE THREAD

**Issue by jellchou** [bytedance/LatentSync#48]
I just modify the 'cuda' device to 'mps', and  decord package was replaced by [(https://pypi.org/project/eva-decord/)]
![1736341101695](https://github.com/user-attachments/assets/854a063a-7689-44bb-9c7d-ebe8ad97331e)

**Comment by jellchou** [comment]
Also in models/resnet.py   I changed the code  from `hidden_states = F.interpolate(hidden_states, scale_factor=[1.0, 2.0, 2.0], mode="nearest")
`      to
` hidden_states = F.interpolate(hidden_states.to(torch.float32), scale_factor=[1.0, 2.0, 2.0], mode="nearest").to(hidden_states.dtype)`

**Comment by jellchou** [comment]
Succeeded  with ComfyUI-LatentSyncWrapper project on M3 Pro!

**Comment by yurisfernandez** [comment]
@jellchou Hey man, I'm going crazy trying to get this to work, would you mind sharing a zip file with the Changes you made in LatentSyncWrapper node?

I am using a macbook pro m4 and it's a nightmare making this run

**Comment by jellchou** [comment]
> I will leave my e m a il here if you want to reach out [yurisfernandez2@gmail.com](mailto:yurisfernandez2@gmail.com)
> 
> Thank you!

The mail has been sent.
